source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,81 @@
 class UrlsController < ApplicationController
+  before_action :set_url, only: [:show, :edit, :update, :destroy]
```

Comment:
another way to do this is to set the callback in the `ApplicationController` and use `skip_before_action` here